model/pytorch/dffml_model_pytorch/utils/utils.py

```diff
@@ -34,6 +33,45 @@ def create_layer(layer_dict):
     return sequential_dict
 
 
+def create_network(network_dict):
+    sequential_network = nn.Sequential()
+    for name, layer in network_dict.items():
+        if isinstance(layer, dict):
```

Comment:
We are checking for a dict object it is so then continue

Reply:
`model2 = {
            "conv1": {
                "layer_type": "Conv2d",
                "in_channels": 3,
                "out_channels": 32,
                "kernel_size": 3,
            },
            "relu1": {"layer_type": "ReLU"},
            "conv2": {
                "layer_type": "Conv2d",
                "in_channels": None,
                "out_channels": 64,
                "kernel_size": 3,
            },
            "relu2": {"layer_type": "ReLU"},
            "conv3": {
                "layer_type": "Conv2d",
                "in_channels": "",
                "out_channels": 96,
                "kernel_size": 3,
            },
            "relu3": {"layer_type": "ReLU"},
            "conv4": {
                "layer_type": "Conv2d",
                "in_channels": "96",
                "out_channels": 32,
                "kernel_size": 3,
            },
        }`

Basicall this function is taking up this kinda models as its parameters

Reply:
In `conv3` the `in_channels` are not set but it is valid as the `conv` layer will be taking its `in_channel` param from the `out_channel` of the previous `conv` layers, so this function is basically handling those stuffs and its converting it to a valid `nn.Sequential`  model

Reply:
[Reference](https://intel.github.io/dffml/master/tutorials/neuralnetwork.html#command-line), a sequential layer is only created when the layers are indented inside a key name like `block1:`, else the layers are added normally (in the model.yaml file you can see how each conv layer is called explicitly in the forward method)

There is a case when there is no sequential model and the layers are feeded to the `create_layer` function one by one. Maybe we should pass the complete model dict in the `pytorch_base.py` file instead of passing layers one by one.

Reply:
Yes but if we pass only the layers one by one ,  we may loose the track of the `out_channels` of the above layers(but it's required as per the issue) .So I thaught of passing the entire `dict`